src/Integrations/Integrations/LaravelQueue/LaravelQueueIntegration.php

```diff
@@ -109,12 +127,28 @@ public function init()
             ]
         );
 
+        hook_method(
+            'Illuminate\Queue\Worker',
+            'maxAttemptsExceededException',
+            null,
+            function ($worker, $scope, $args, $retval) use ($integration) {
+                if (($rootSpan = \DDTrace\root_span()) !== null) {
+                    $integration->setError($rootSpan, $retval);
+                }
+            }
+        );
+
         trace_method(
             'Illuminate\Queue\Jobs\Job',
             'fire',
-            function (SpanData $span, $args, $retval, $exception) use ($integration) {
-                $integration->setSpanAttributes($span, 'laravel.queue.fire', 'process', $this, $exception);
-            }
+            [
+                'prehook' => function (SpanData $span, $args, $retval) use ($integration) {
+                    $integration->setSpanAttributes($span, 'laravel.queue.fire', 'process', $this);
+                },
+                'posthook' => function (SpanData $span, $args, $retval, $exception) use ($integration) {
+                    $integration->setSpanAttributes($span, 'laravel.queue.fire', 'process', $this, $exception);
```

Comment:
```suggestion
                    if ($exception) {
                        $this->setError($span, $exception);
                    }
```
Avoid duplicating setting it :-)